torch/csrc/jit/passes/guard_elimination.cpp

```diff
@@ -273,6 +273,21 @@ struct GuardElimination {
     case aten::reciprocal:
     case aten::addcmul:
     case aten::where:
+    case aten::softmax:
+    case aten::log_softmax:
+    case aten::log_sigmoid:
+    case aten::dropout:
+    case aten::matmul:
+    case aten::bmm:
+    case aten::to:
```

Comment:
I don't think this is right. `checkInputs()` checks if all inputs are summarized, or constant, or a subtype of Number. ScalarTypes are currently represented as an Integer, it is a subtype of Number. That means if you have a variable dtype which is `int` the first 10 iterations, but then `float` the next after, this pass will incorrectly remove the guard on the output being an integer tensor.

Reply:
I would imagine this is also possible with some of the other cases that are in guard elimination.

Reply:
I'll remove `aten::to`.  Do you have any others that you think are incorrect?  To me it looks like the rest of them should be OK?

Reply:
anything that has a `dtype` or `layout` argument, probably. i know at least `aten::rand_like`. one short term fix is to require a constant value if an integer type input the name `dtype` or `layout`. cc @Krovatkin .

Reply:
agreed. i'm going to add `allow_numbers = false` to `checkInputs` or maybe remove `||a subtype of Number` altogether, then the cases like `clamp` can be just implemented with exceptions on specific inputs, but in a short term, `allow_numbers` might be a bit more practical.